i6_models/util/compat.py

```diff
@@ -12,7 +12,7 @@ def logical_not(tensor: torch.Tensor, /) -> torch.Tensor:
 
     :param tensor: bool mask of shape (B, T) to be inverted.
     """
-    if torch.onnx.is_in_onnx_export():
+    if torch.jit.is_scripting() or torch.jit.is_tracing():
```

Comment:
does this include all scenarios where `is_in_onnx_export` is `True`?

Reply:
I thought it is, but going through the code I am actually not so sure anymore. It seems the internal tracing via the ONNX call is not exactly the same as the regular `jit.trace()`. I think just to be safe we might want to have all 3 checks....

Reply:
Why is this change needed for ONNX export?

The code below (`logical_not`) was specifically about missing (or buggy/broken) Torch-to-ONNX support, not about scripting or tracing.

Reply:
You might want to use e.g. scripting instead of tracing, but then this is called before the actual ONNX export, so `is_in_onnx_export` does not work.

Reply:
You might want to use scripting or tracing and not use ONNX, and then you would not want to have this, so this condition would be wrong then.

We should just make it more explicit then. We can have an own global flag `in_onnx_export`, which we can set in our torch-to-onnx-export script.

Reply:
I will just revert this change for now to not delay this further, scripting is anyway something only I used so far. Then it works as expected for the RETURNN script, and I will modify my own setup.